### PR TITLE
Embed GeoJSON schema in the project to improve build reliability

### LINKED ISSUE #125: GeoJSON schema
Currently the GeoJSON schema is fetched at build time at https://geojson.org/schema/GeoJSON.json.

There are drawbacks at doing this:
- the project cannot be build offline
- there is no version of the schema in the URL (potential risk of breaking change)

A simple solution could be to embed it in the project, in the current version.

### PR DESCRIPTION
**Pull request by arjxn-py**
Should fix #125

### CODE REVIEW

**Comment by martinRenou** [comment]
Thanks for opening a PR! Although I kind of disagree with the fix.

I'd be more in favor of an approach to:
- Download the schema once
- Store it somewhere that is not cleaned with `jlpm run clean` and not tracked by git
- Don't download it again if it's already there

That way you need an internet connection for the first build, but you don't need it anymore for subsequent builds.

**Comment by arjxn-py** [comment]
Thanks @martinRenou, I agree with your idea and just want to confirm a couple of things that - 
- Do we want to download the schema once while building the extension if that's not already there? 
- Would this implementation be something similar like thumbnails for raster layers thing?

**Comment by martinRenou** [comment]
> Do we want to download the schema once while building the extension (or while loading geojson layer) if base schema is not already there?

Once when building the application (we need the schemas to generate some Python interfaces as of today).

> Would this implementation be something similar like thumbnails for raster layers thing?

Yes that could be similar 👍🏽

**Comment by arjxn-py** [comment]
Oh oops i pushed in wrong branch

File: packages/schema/cacheGeoJSONSchema.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const fs = require('fs');
+const https = require('https');
+const path = require('path');
+
+const schemaUrl = 'https://geojson.org/schema/GeoJSON.json';
+const cacheDir = path.resolve(__dirname, '.cache/schemas');
```
Comment (martinRenou) [inline packages/schema/cacheGeoJSONSchema.js:6]: Should we add an entry for `.gitignore` for this?

File: packages/schema/package.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
   "scripts": {
     "build": "jlpm build:schema && jlpm build:lib",
     "build:schema": "jlpm build:schema:js && jlpm build:schema:py",
```
Comment (martinRenou) [inline packages/schema/package.json:29]: Shouldn't it be this instead? Downloading and caching geojson schema before generating interfaces for TS and Python?

```suggestion
    "build:schema": "node ./cacheGeoJSONSchema.js && jlpm build:schema:js && jlpm build:schema:py",
```

Comment (arjxn-py) [reply packages/schema/package.json:29]: Ahh yes, i didn't realize

File: packages/schema/cacheGeoJSONSchema.js (hunk 1 above)
Comment (arjxn-py) [reply packages/schema/cacheGeoJSONSchema.js:6]: We don't need to, since folder name is `.cache` and it's already there

Comment (martinRenou) [reply packages/schema/cacheGeoJSONSchema.js:6]: ok!

**Review by mfisher87 — APPROVED** [review]

File: packages/schema/package.json [hunk 3]
```diff
@@ -26,8 +26,8 @@
   },
   "scripts": {
     "build": "jlpm build:schema && jlpm build:lib",
-    "build:schema": "jlpm build:schema:js && jlpm build:schema:py",
-    "build:schema:js": "json2ts -i src/schema -o src/_interface --no-unknownAny --unreachableDefinitions --cwd ./src/schema && cd src/schema && node ../../schema.js",
+    "build:schema": "node ./cacheGeoJSONSchema.js && jlpm build:schema:js && jlpm build:schema:py",
+    "build:schema:js": "json2ts -i src/schema -o src/_interface --no-unknownAny --unreachableDefinitions --cwd ./src/schema && cd src/schema && node ../../schema.js && node ../../cacheGeoJSONSchema.js",
```
Comment (martinRenou) [inline packages/schema/package.json:30]:
```suggestion
    "build:schema:js": "json2ts -i src/schema -o src/_interface --no-unknownAny --unreachableDefinitions --cwd ./src/schema && cd src/schema && node ../../schema.js",
```

**Review by martinRenou — CHANGES_REQUESTED** [review]

**Review by martinRenou — APPROVED** [review]
Thanks!

**Pull request merged**